src/terminal/input/terminalInput.cpp

```diff
@@ -284,93 +332,45 @@ bool TerminalInput::_SearchWithModifier(const KeyEvent& keyEvent) const
-                                      const size_t cKeyMapping,
-                                      _Out_ const TerminalInput::_TermKeyMap** pMatchingMapping) const
-{
-    bool fKeyTranslated = false;
-    for (size_t i = 0; i < cKeyMapping; i++)
-    {
-        const _TermKeyMap* const pMap = &(keyMapping[i]);
-
-        if (pMap->wVirtualKey == keyEvent.GetVirtualKeyCode())
-        {
-            // If the mapping has no modifiers set, then it doesn't really care
-            //      what the modifiers are on the key. The caller will likely do
-            //      something with them.
-            // However, if there are modifiers set, then we only want to match
-            //      if the key's modifiers are the same as the modifiers in the
-            //      mapping.
-            bool modifiersMatch = WI_AreAllFlagsClear(pMap->dwModifiers, MOD_PRESSED);
-            if (!modifiersMatch)
-            {
-                // The modifier mapping expects certain modifier keys to be
-                //      pressed. Check those as well.
-                modifiersMatch =
-                    (WI_IsFlagSet(pMap->dwModifiers, SHIFT_PRESSED) == keyEvent.IsShiftPressed()) &&
-                    (WI_IsAnyFlagSet(pMap->dwModifiers, ALT_PRESSED) == keyEvent.IsAltPressed()) &&
-                    (WI_IsAnyFlagSet(pMap->dwModifiers, CTRL_PRESSED) == keyEvent.IsCtrlPressed());
-            }
-
-            if (modifiersMatch)
-            {
-                fKeyTranslated = true;
-                *pMatchingMapping = pMap;
-                break;
-            }
-        }
-    }
-    return fKeyTranslated;
+    return success;
 }
 
 // Routine Description:
 // - Searches the input array of mappings, and sends it to the input if a match was found.
 // Arguments:
 // - keyEvent - Key event to translate
 // - keyMapping - Array of key mappings to search
-// - cKeyMapping - number of entries in keyMapping
+// - sender - Function to use to dispatch translated event
 // Return Value:
 // - True if there was a match to a key translation, and we successfully sent it to the input
-bool TerminalInput::_TranslateDefaultMapping(const KeyEvent& keyEvent,
-                                             _In_reads_(cKeyMapping) const TerminalInput::_TermKeyMap* keyMapping,
-                                             const size_t cKeyMapping) const
+bool _TranslateDefaultMapping(const KeyEvent& keyEvent,
+                              const std::basic_string_view<TermKeyMap> keyMapping,
+                              InputSender sender)
 {
-    const TerminalInput::_TermKeyMap* pMatchingMapping;
-    const bool fSuccess = _SearchKeyMapping(keyEvent, keyMapping, cKeyMapping, &pMatchingMapping);
-    if (fSuccess)
+    const auto match = _searchKeyMapping(keyEvent, keyMapping);
+    if (match)
     {
-        _SendInputSequence(pMatchingMapping->pwszSequence);
+        sender(match.value().sequence);

```

Comment:
use of optional: you can use `->` instead of `.value().` because you already checked its residence

Reply:
unless SA croaks

Reply:
nah its fine, good call. thanks.